libgit/worktree_storer.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"github.com/pkg/errors"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/index"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+)
+
+// worktreeStorer fake out the storage layer in such a way that it
+// stores its index, and any reference modifications, in a .git/
+// directory in the worktree, but everything else goes through the
+// main bare repo's .git directory.  This is useful for making a quick
+// checkout of a git repo in a new directory, without making a full
+// git clone. For example, consider the following instance:
+//
+// * `storage.Storer` points to a bare repo stored in KBFS, say
+//   `/keybase/team/keybase/.kbfs_git/secrets/.git`.
+// * `delta` is the same as above, but cast as a `DeltaObjectStorer`.
+// * `wtDotgit` can be pointed anywhere else in KBFS where you want to
+//   expose a checked-out version of the keybase secrets repo, say
+//   `/keybase/private/strib/.kbfs_autogit/team/keybase/secrets/`
```

Comment:
In the `repoFromStorageAndWorktree` [above](https://github.com/keybase/kbfs/pull/1400/files#diff-e3df4596a1bff828fb4e958580cd23b3R54), `wtDotgit` is the ".git" subdir of the worktree. So it looks like this should've been `/keybase/private/strib/.kbfs_autogit/team/keybase/secrets/.git`?